go/test/endtoend/sharding/verticalsplit/vertical_split_test.go

```diff
@@ -0,0 +1,766 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package verticalsplit
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"vitess.io/vitess/go/json2"
+	"vitess.io/vitess/go/vt/vtgate/vtgateconn"
+
+	"github.com/stretchr/testify/assert"
+	"vitess.io/vitess/go/mysql"
+	"vitess.io/vitess/go/sqltypes"
+	"vitess.io/vitess/go/test/endtoend/cluster"
+	"vitess.io/vitess/go/test/endtoend/sharding"
+	"vitess.io/vitess/go/vt/proto/topodata"
+	_ "vitess.io/vitess/go/vt/vtgate/grpcvtgateconn"
+)
+
+var (
+	clusterInstance      *cluster.LocalProcessCluster
+	sourceKeyspace       = "source_keyspace"
+	destinationKeyspace  = "destination_keyspace"
+	hostname             = "localhost"
+	cellj                = "test_nj"
+	shardName            = "0"
+	createTabletTemplate = `
+		create table %s(
+			id bigint not null,
+			msg varchar(64),
+			primary key (id),
+			index by_msg (msg)
+		) Engine=InnoDB;`
+	createViewTemplate     = "create view %s(id, msg) as select id, msg from %s;"
+	createMoving3NoPkTable = `
+		create table moving3_no_pk (
+			id bigint not null,
+			msg varchar(64)
+		) Engine=InnoDB;`
+	tableArr         = []string{"moving1", "moving2", "staying1", "staying2"}
+	insertIndex      = 0
+	moving1First     int
+	moving2First     int
+	staying1First    int
+	staying2First    int
+	moving3NoPkFirst int
+)
+
+func TestVerticalSplit(t *testing.T) {
+	// setting RBR mode
+	os.Setenv("EXTRA_MY_CNF", path.Join(os.Getenv("VTROOT"), "config", "mycnf", "rbr.cnf"))
+	// setting RBR mode
+
+	flag.Parse()
+	code, err := initializeCluster(t)
+	if err != nil {
+		t.Errorf("setup failed with status code %d", code)
+	}
+	defer teardownCluster(t)
+
+	// Adding another cell in the same cluster
+	err = clusterInstance.TopoProcess.ManageTopoDir("mkdir", "/vitess/"+"test_ca")
+	err = clusterInstance.VtctlProcess.AddCellInfo("test_ca")
+	err = clusterInstance.TopoProcess.ManageTopoDir("mkdir", "/vitess/"+"test_ny")
+	err = clusterInstance.VtctlProcess.AddCellInfo("test_ny")
+
+	assert.Nil(t, err, "error should be Nil")
+
+	// source keyspace, with 4 tables
+	sourceShard := clusterInstance.Keyspaces[0].Shards[0]
+	sourceMasterTablet := sourceShard.Vttablets[0]
+	sourceReplicaTablet := sourceShard.Vttablets[1]
+	sourceRdOnlyTablet1 := sourceShard.Vttablets[2]
+	sourceRdOnlyTablet2 := sourceShard.Vttablets[3]
+	sourceKs := fmt.Sprintf("%s/%s", sourceKeyspace, shardName)
+
+	// master tablet init
+	_ = clusterInstance.VtctlclientProcess.InitTablet(&sourceMasterTablet, cellj, sourceKeyspace, hostname, sourceShard.Name)
+	_ = sourceMasterTablet.VttabletProcess.CreateDB(sourceKeyspace)
+
+	// replica tablet init
+	_ = clusterInstance.VtctlclientProcess.InitTablet(&sourceReplicaTablet, cellj, sourceKeyspace, hostname, sourceShard.Name)
+	_ = sourceReplicaTablet.VttabletProcess.CreateDB(sourceKeyspace)
+
+	// rdOnly_1 tablet init
```

Comment:
in vitess lingo rdonly is one word. Let's also stop using `_`, that is python style

Reply:
Done. Refer [PR46](https://github.com/planetscale/vitess/pull/46)